File: SearchView(java)/app/src/main/java/coder/mtk/newappjava/api/ApiInterface.java
package coder.mtk.newappjava.api;

import coder.mtk.newappjava.model.News;
import retrofit2.Call;
import retrofit2.Response;
import retrofit2.http.GET;
import retrofit2.http.Query;

interface ApiInterface {

    @GET("top-headlines")
    Call<News> getNews(
            @Query("country") String country,
            @Query("category") String category,
            @Query("apiKey") String apiKey
    );
}

File: Json(retrofitSample)/app/src/main/java/coder/mtk/jsonretrofitsample/api/ApiInterface.kt
package coder.mtk.jsonretrofitsample.api

import coder.mtk.jsonretrofitsample.model.PostItem
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path

interface ApiInterface {
        @GET("posts")
    fun getPost(): Call<ArrayList<PostItem>>

    @GET("posts/{postID}")
    fun getPostWidthID(
        @Path("postID")postID : String
    ) : Call<PostItem>
}